renderdoc/driver/vulkan/vk_debug.cpp

```diff
@@ -2741,24 +2741,43 @@ void VulkanDebugManager::ReplaceResource(ResourceId from, ResourceId to)
 
     if(refdShader)
     {
-      VkGraphicsPipelineCreateInfo pipeCreateInfo;
-      MakeGraphicsPipelineInfo(pipeCreateInfo, it->first);
+      VkPipeline pipe = VK_NULL_HANDLE;
+      const VulkanCreationInfo::Pipeline &pipeInfo = m_pDriver->m_CreationInfo.m_Pipeline[it->first];
+      if(pipeInfo.renderpass.id)    // check if this is a graphics or compute pipeline
+      {
+        VkGraphicsPipelineCreateInfo pipeCreateInfo;
+        MakeGraphicsPipelineInfo(pipeCreateInfo, it->first);
 
-      // replace the relevant module
-      for(uint32_t i = 0; i < pipeCreateInfo.stageCount; i++)
+        // replace the relevant module
+        for(uint32_t i = 0; i < pipeCreateInfo.stageCount; i++)
+        {
+          VkPipelineShaderStageCreateInfo &sh =
+              (VkPipelineShaderStageCreateInfo &)pipeCreateInfo.pStages[i];
+
+          if(sh.module == srcShaderModule)
+            sh.module = dstShaderModule;
+        }
+
+        // create the new graphics pipeline
+        VkResult vkr = m_pDriver->vkCreateGraphicsPipelines(dev, VK_NULL_HANDLE, 1, &pipeCreateInfo,
+                                                            NULL, &pipe);
+        RDCASSERTEQUAL(vkr, VK_SUCCESS);
+      }
+      else
       {
-        VkPipelineShaderStageCreateInfo &sh =
-            (VkPipelineShaderStageCreateInfo &)pipeCreateInfo.pStages[i];
+        VkComputePipelineCreateInfo pipeCreateInfo;
+        MakeComputePipelineInfo(pipeCreateInfo, it->first);
 
+        // replace the relevant module
+        VkPipelineShaderStageCreateInfo &sh = pipeCreateInfo.stage;
```

Comment:
I guess this check is redundant since there's only one module. Maybe just demote this to an `RDCASSERT()`?